File: backend/Testing/Browser/Page/ForgotPasswordPage.cs
using Microsoft.Playwright;

namespace Testing.Browser.Page;

public class ForgotPasswordPage : BasePage<ForgotPasswordPage>
{
    public ForgotPasswordPage(IPage page)
    : base(page, "/forgotPassword", page.GetByRole(AriaRole.Heading, new() { Name = "Forgot Password" }))
    {
    }

    public async Task FillForm(string email)
    {
        await Page.GetByLabel("Email").FillAsync(email);
    }

    public async Task<ResetPasswordEmailSentPage> Submit()
    {
        await Page.GetByRole(AriaRole.Button, new() { Name = "Send reset email" }).ClickAsync();
        return await new ResetPasswordEmailSentPage(Page).WaitFor();
    }
}

public class ResetPasswordEmailSentPage : BasePage<ResetPasswordEmailSentPage>
{
    public ResetPasswordEmailSentPage(IPage page)
    : base(page, "/forgotPassword/emailSent", page.GetByRole(AriaRole.Heading, new() { Name = "Check Your Inbox" }))
    {
    }
}

File: frontend/tests/pages/forgotPasswordPage.ts
import type { Page } from '@playwright/test';
import { BasePage } from './basePage';

export class ResetPasswordEmailSentPage extends BasePage {
  constructor(page: Page) {
    super(page, page.getByRole('heading', {name: 'Check Your Inbox'}), '/forgotPassword/emailSent');
  }
}

export class ForgotPasswordPage extends BasePage {
  constructor(page: Page) {
    super(page, page.getByRole('heading', {name: 'Forgot Password'}), '/forgotPassword');
  }

  async fillForm(email: string): Promise<void>
  {
      await this.page.getByLabel('Email').fill(email);
  }

  async submit(): Promise<ResetPasswordEmailSentPage>
  {
      await this.page.getByRole('button', { name: 'Send reset email' }).click();
      return await new ResetPasswordEmailSentPage(this.page).waitFor();
    }
}
